### ISSUE TITLE
jsx-a11y/no-static-element-interactions does not work with variables vs hardcoded strings for role value

### ISSUE THREAD

**Issue by bradjboose** [jsx-eslint/eslint-plugin-jsx-a11y#689]
Hello---

I am getting errors from `jsx-a11y/no-static-element-interactions ` on my components ONLY when using a variable for the role name vs a hardcoded string.

This passes the linting:

```
<div
  tabIndex={0}
  onClick={onWrapperClick}
  onKeyDown={onWrapperKeyDown}
  role='button'
>
....
```

However, this does NOT pass:

```
<div
  tabIndex={0}
  onClick={onWrapperClick}
  onKeyDown={onWrapperKeyDown}
  role={AriaRoles.BUTTON}
>
....
```

The values for `AriaRoles.BUTTON` is 'button'. It seems the library cannot tell that the variable it a valid string...

**Comment by ljharb** [comment]
This is intentional; it must be a hardcoded string. Static analysis - linters - are unable to follow references to look up dynamic runtime values.

**Comment by bradjboose** [comment]
Gotcha. Thanks for the swift response.
